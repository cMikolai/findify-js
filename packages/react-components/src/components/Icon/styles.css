.root{
  display: inline-block;
  box-sizing: content-box;
  /*color: inherit; FIXME: for whatever reason styles override each other while ignoring class order completely*/
}
