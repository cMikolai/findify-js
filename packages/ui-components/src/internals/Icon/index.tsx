import * as React from 'react';
import { compose, defaultProps, mapProps, componentFromProp } from 'recompose';

export interface OwnProps {
  name: string;
  width?: number;
  height?: number;
}

export type MappedProps = {
  width: number;
  height: number;
};

export type Props = OwnProps & MappedProps;

const svgLoader = '!!babel-loader!react-svg-loader!./icons/';
const getIcon = (name: string) => require(`${svgLoader}${name}.svg`).default;

const Icon = compose<OwnProps, Props>(
  defaultProps<MappedProps>({
    width: 16,
    height: 16,
  }),
  mapProps<OwnProps, Props>((props: Props) => ({
    width: props.width,
    height: props.height,
    component: getIcon(props),
  }))
)(componentFromProp('component'));

export default Icon;
