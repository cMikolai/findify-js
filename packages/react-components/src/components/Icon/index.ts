import React from 'react';
import cx from 'classnames';
import styles from 'components/Icon/styles.css';

import Filters from 'components/Icon/icons/Filters.svg';
import Minus from 'components/Icon/icons/Minus.svg';
import Plus from 'components/Icon/icons/Plus.svg';
import Search from 'components/Icon/icons/Search.svg';
import Sorting from 'components/Icon/icons/Sorting.svg';

import CheckmarkDark from 'components/Icon/icons/Checkmark/Dark.svg';
import Star from 'components/Icon/icons/Star.svg';
import XDark from 'components/Icon/icons/X/Dark.svg';
import XMobile from 'components/Icon/icons/X/Mobile.svg';
import RadioEmpty from 'components/Icon/icons/Radio/Empty.svg';
import RadioFilled from 'components/Icon/icons/Radio/Filled.svg';
import ArrowBack from 'components/Icon/icons/Arrow/Back.svg';
import ArrowDown from 'components/Icon/icons/Arrow/Down.svg';
import ArrowUp from 'components/Icon/icons/Arrow/Up.svg';
import ArrowLeft from 'components/Icon/icons/Arrow/Left.svg';
import ArrowRight from 'components/Icon/icons/Arrow/Right.svg';
import ArrowLeftBig from './icons/Arrow/Left-Big.svg';
import ArrowRightBig from './icons/Arrow/Right-Big.svg';
import CheckboxFilled from 'components/Icon/icons/Checkbox/Filled.svg';
import CheckboxEmpty from 'components/Icon/icons/Checkbox/Empty.svg';

const icons = {
  Filters,
  Minus,
  Plus,
  Search,
  Sorting,
  CheckmarkDark,
  Star,
  XDark,
  XMobile,
  RadioEmpty,
  RadioFilled,
  ArrowBack,
  ArrowDown,
  ArrowUp,
  ArrowLeft,
  ArrowRight,
  ArrowLeftBig,
  ArrowRightBig,
  CheckboxFilled,
  CheckboxEmpty
}

export type Props = {
  name: keyof typeof icons
  width?: number
  height?: number
  className?: string
}

export default ({ name, width, height, className, ...rest }: Props) =>
  React.createElement(icons[name], {
    width,
    height,
    className: cx(styles.root, className),
    ...rest
  })

