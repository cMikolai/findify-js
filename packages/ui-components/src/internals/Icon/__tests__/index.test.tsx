import * as React from 'react';
import { shallow, render } from 'enzyme';
import Icon from '..';

describe('<Icon />', () => {
  it('renders correctly', () => {
    // TODO: read https://diessi.ca/blog/how-to-exclude-css-images-anything-from-unit-tests/
    // const component = render(<Icon name="cross" />);
    // expect(component).toMatchSnapshot();
  });
});
