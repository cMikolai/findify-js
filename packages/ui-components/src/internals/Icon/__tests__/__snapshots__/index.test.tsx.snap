// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> attaches and calls onClick event handler 1`] = `
<mapProps(componentFromProp(component))
  height={18}
  name="cross"
  onClick={[MockFunction]}
  width={15}
/>
`;

exports[`<Icon /> renders correctly 1`] = `
<mapProps(componentFromProp(component))
  height={16}
  name="cross"
  width={16}
/>
`;

exports[`<Icon /> renders correctly 2`] = `
<mapProps(componentFromProp(component))
  height={40}
  name="arrow"
  width={32}
/>
`;

exports[`<Icon /> renders correctly 3`] = `
<mapProps(componentFromProp(component))
  height={20}
  name="check"
  style={
    Object {
      "padding": 4,
    }
  }
  width={12}
/>
`;

exports[`<Icon /> renders correctly 4`] = `
<mapProps(componentFromProp(component))
  className="foo"
  height={16}
  name="cross"
  width={16}
/>
`;

exports[`<Icon /> renders correctly 5`] = `
<mapProps(componentFromProp(component))
  className="foo"
  height={16}
  name="cross"
  style={
    Object {
      "display": "none",
    }
  }
  width={16}
/>
`;
